plugins/lightspeed-backend/src/service/router.ts

```diff
@@ -145,8 +164,18 @@ export async function createRouter(
         }
         response.end();
 
-        await saveHistory(conversation_id, Roles.HumanRole, query);
-        await saveHistory(conversation_id, Roles.AIRole, content);
+        await saveHistory(
+          conversation_id,
+          Roles.HumanRole,
+          query,
+          userMessageTimestamp,
```

Comment:
can we also simply test the timestamp as well? probably just check the query response contains a timestamp, and it's increased compare with previous messages.